daemon/daemon.go

```diff
@@ -86,12 +86,20 @@ type Command struct {
 	d *Daemon
 }
 
+type accessResult int
+
+const (
+	accessOK accessResult = iota
+	accessUnauthorized
+	accessForbidden
+)
```

Comment:
a different approach would be to have canAccess return an error directly,   so no error is OK and the other cases return the needed errors. Given that canAccess is used in one place it would work.  If we used canAccess to return sometimes partial results from some read queries then this approach maybe would be better.  I don't have a strong preference atm.